datadog/dogstatsd/metrics.py

```diff
@@ -0,0 +1,77 @@
+from threading import Lock
+
+
+class MetricAggregator(object):
+    def __init__(self, name, tags, rate):
+        self.name = name
+        self.tags = tags
+        self.rate = rate
+
+    def aggregate(self, value):
+        raise NotImplementedError("Subclasses should implement this method.")
+
+    def flush_unsafe(self):
+        raise NotImplementedError("Subclasses should implement this method.")
+
+
+class CountMetric(MetricAggregator):
+    def __init__(self, name, value, tags, rate):
+        super(CountMetric, self).__init__(name, tags, rate)
+        self.value = value
+
+    def aggregate(self, v):
+        self.value += v
+
+    def flush_unsafe(self):
+        return {
+            "metric_type": "count",
+            "name": self.name,
+            "tags": self.tags,
+            "rate": self.rate,
+            "ivalue": self.value,
+        }
+
+
+class GaugeMetric(MetricAggregator):
+    def __init__(self, name, value, tags, rate):
+        super(GaugeMetric, self).__init__(name, tags, rate)
+        self.value = value
+
+    def aggregate(self, v):
+        self.value = v
+
+    def flush_unsafe(self):
+        return {
```

Comment:
Would it be better to create a new class for this object? Not too familiar with python conventions

Reply:
<img width="583" alt="image" src="https://github.com/DataDog/datadogpy/assets/171708865/fe63c68f-47ef-4803-9a53-e816d0f44b78">

I was thinking that the purpose of having a Metric struct was so that they can be grouped into one list, since we can already do that in python I thought it wasn't necessary.

Reply:
Maybe we can omit the `flush` function for now since the current implementation of `flush_unsafe` makes assumptions about the structure of the data in regard to how it will be serialized and flushed before that part of the code is written. 

I'd consider making these functions `TODO` for now and introduce the flush data structure in the next PR. It's hard to argue which approach is better without seeing how it will ultimately be used. In fact - we may not even need these flush functions depending on how the metric classes are going to be used.

For example - I wonder if it makes sense to move metric serialization as a function on each metric class? So you can do something like:

```python
countMetric = CountMetric()
flush(countMetric.serialize())
```